src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -57,6 +57,15 @@ internal TagHelperExecutionContext(string tagName)
         }
 
         /// <summary>
+        /// Indicates whether or not the tag is self closing.
+        /// </summary>
+        public bool SelfClosing
```

Comment:
@dougbu fair point, We probably shouldn't have put that on the `TagHelperExecutionContext` in hindsight. It's only there to output what happened from the result of running the `TagHelper`s. Ultimately we should have created a helper class to handle that responsibility.